dotcom-rendering/src/web/components/HeaderTopBarSearchCapi.tsx

```diff
@@ -0,0 +1,211 @@
+import {
+	brand,
+	brandAlt,
+	from,
+	neutral,
+	textSans,
+} from '@guardian/source-foundations';
+import { useEffect, useRef, useState } from 'react';
+import SearchIcon from '../../static/icons/search.svg';
+import { getZIndex } from '../lib/getZIndex';
+import { useApi } from '../lib/useApi';
+
+const searchLinkStyles = css`
+	${textSans.medium({ fontWeight: 'bold' })};
+	color: ${neutral[100]};
+	transition: color 80ms ease-out;
+	text-decoration: none;
+	padding: 7px 0;
+	background: none;
+	border: none;
+
+	${from.tablet} {
+		padding: 7px 10px 7px 5px;
+	}
+
+	:hover,
+	:focus {
+		text-decoration: underline;
+	}
+
+	svg {
+		fill: currentColor;
+		float: left;
+		height: 18px;
+		width: 18px;
+		margin: 3px 4px 0 0;
+	}
+	${getZIndex('searchHeaderLink')}
+`;
+
+const linkTablet = css`
+	display: none;
+	position: relative;
+
+	:before {
+		content: '';
+		border-left: 1px solid ${brand[600]};
+		height: 24px;
+	}
+
+	${from.desktop} {
+		display: flex;
+	}
+`;
+
+export const HeaderTopBarSearchCapi = () => {
+	const [isSearchVisible, setIsSearchVisible] = useState(false);
+	const [q, setQ] = useState<string>();
+	const qRef = useRef<HTMLInputElement>(null);
+	const { data } = useApi<{
+		response: { results: { webTitle: string; webUrl: string }[] };
+	}>(
+		q
+			? `https://content.guardianapis.com/search?api-key=test&q=${q}`
+			: undefined,
+	);
+
+	const formRef = useRef<HTMLFormElement>(null);
+
+	// Hide search if you click anywhere outside of search when it's shown
+	useEffect(() => {
+		function handleClickOutside(event: MouseEvent) {
+			if (
+				!formRef.current?.contains(event.target as Node) &&
+				isSearchVisible
+			) {
+				setQ(undefined);
+				setIsSearchVisible(false);
+			}
+		}
+
+		document.addEventListener('mousedown', handleClickOutside);
+		return () => {
+			document.removeEventListener('mousedown', handleClickOutside);
+		};
+	}, [formRef, isSearchVisible]);
+
+	// Hide search if you press escape and it's show
+	useEffect(() => {
+		const hideSearch = (event: KeyboardEvent) => {
+			if (event.key === 'Escape') {
+				setQ(undefined);
+				setIsSearchVisible(false);
+			}
+		};
+		document.addEventListener('keydown', hideSearch, false);
+		return () => {
+			document.removeEventListener('keydown', hideSearch, false);
+		};
+	}, []);
+
+	// Focus the input element when we show search
+	useEffect(() => {
+		if (isSearchVisible) {
+			qRef.current?.focus();
+		}
+	}, [isSearchVisible]);
+
+	return (
+		<>
+			<div css={linkTablet}>
+				<a
+					href="https://www.google.co.uk/advanced_search?q=site:www.theguardian.com"
+					type="button"
+					css={searchLinkStyles}
+					data-link-name="nav3 : search"
+					onClick={(event) => {
+						event.preventDefault();
+
+						// toggle search
+						if (isSearchVisible) {
+							setQ(undefined);
+							setIsSearchVisible(false);
+						} else {
+							setIsSearchVisible(true);
+						}
+
+						return false;
+					}}
+				>
+					<SearchIcon />
+					<>Search</>
```

Comment:
Do you really need to wrap that?

Reply:
Copy/pasta from https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/components/HeaderTopBarSearch.tsx#L56